type/__download/gencode-remote

```diff
@@ -17,7 +17,7 @@ then
         dst="/$__object_id"
     fi
 
-    echo "download_tmp=\"\$( mktemp )\""
+    echo "download_tmp=\"\$( mktemp '$dst'.XXXXXXXX )\""
```

Comment:
This makes totally sense for remote, but can we actually drop `mktemp` and do `echo "download_tmp=\"$dst.cdist__download_tmp\""` instead?

Reply:
Using `.cdist__download_tmp` sounds like a reasonable choice to me.
This also has the advantage that moving to the real destination won't require a copy if the tmpdir is on a separate file system.

Unless I'm overlooking something it doesn't make sense to download multiple things to the same destination at all.
If this should be a valid use case, maybe we could use the random ID of the cdist object (`${object##.cdist-}`)?

Reply:
As the type doesn't allow multiple downloads for a single file (as the object id must be unique), there is no problem here. It will also help not having dangling tempfiles from previously failed tempfiles, as it's now determinant. The `mktemp` is a relic from the original code.